app/game/board.js

```diff
@@ -1,459 +1,354 @@
-*/
-
-//  I used this method to make sure that each intersection only exists once, even 
-//  If it is associated with multiple tiles.  This makes management of settlements, 
-//  roads and cities easier later on
-Board.prototype.findNode = function(nodeIndex, theTile, row, col) {
-    var isEven = ((row % 2) == 0);
-    var theNode = {};
-
-    var newRow = -1;
-    var newCol = -1;
-    var newNode = -1;
-
-    //  Nodes 0-1, 11
-    if (row > 0 && (nodeIndex < 2 || nodeIndex == 11)) {
-        newRow = row - 1;
-        newCol = (nodeIndex == 11 ? (isEven ? col : col - 1) : (isEven ? col + 1 : col));
-        newNode = (nodeIndex == 0 ? 8 : (nodeIndex == 1 ? 7 : 5));
-    }
-
-    //  Nodes 2-4
-    if (col < 6 && (nodeIndex > 1 && nodeIndex < 5)) {
-        newRow = row;
-        newCol = col + 1;
-        newNode = (nodeIndex == 2 ? 10 : (nodeIndex == 3 ? 9 : 8));
-    }
+/// returns node object
+Board.prototype.get_node = function (index) {
+  return this.nodes[index];
+}
 
-    //  Node 5-8
-    if (row < 6 && (nodeIndex > 4 && nodeIndex < 9)) {
-        newRow = row + 1;
-        newCol = (nodeIndex == 5 ? (isEven ? (col < 6 ? col + 1 : -1) : col) : (isEven ? col : (col > 0 ? col - 1 : -1)));
-        newNode = (nodeIndex == 5 ? 11 : (nodeIndex == 6 ? 2 : (nodeIndex == 7 ? 1 : 0)));
-    }
+/// returns tile object
+Board.prototype.get_tile = function (point) {
+  return this.tiles[point.y][point.x];
+};
+
+/// returns resource type string
+Board.prototype.get_tile_resource_type = function (point) {
+  return this.tiles[point.y][point.x].type;
+};
+
+/********************************************
+*  More advanced getters for board elements
+*********************************************/
+/// returns the index numbers for nodes
+Board.prototype.get_node_indexes_from_road = function (index) {
+  return [this.nodes[this.roads[index].connects[0]],
+          this.nodes[this.roads[index].connects[1]]];
+}
 
-    //  Node 9-10
-    if (col > 0 && (nodeIndex == 9 || nodeIndex == 10)) {
-        newRow = row;
-        newCol = col - 1;
-        newNode = (nodeIndex == 9 ? 3 : 2);
+/// returns an array of tile objects
+Board.prototype.get_tiles_with_resource = function (resource) {
+  var array = [];
+  for (var y=0; y< this.tiles.length; y++) {
+    for (var x=0; x< this.tiles.length; x++) {
+      var tile = this.tiles[y][x];
+      if (tile.type === resource)
+        array.push(tile);
     }
-
-    if (newRow >= 0 && newCol >= 0 && newNode >= 0) {
-        theNode = this.nodes[this.board[newRow][newCol].nodes[newNode]];
-        //theNode = this.board[newRow][newCol].nodes[newNode];
+  }
+  return array;
+};
+
+/// returns an array of player names
+Board.prototype.get_players_with_resource = function (resource) {
+  var players = [];
+  var tiles = this.get_tiles_of_resource(resource);
+  for (var t=0; t<this.tiles.length; t++) {
+    if (this.tiles[t].owner)
+      players.push(this.tiles[t].owner);
+  }
+  return players;
+};
```

Comment:
Something not quite right here I'm getting a linter warning saying `tiles` is defined but not used. Should line 249 be deleted since you're using the tiles property on the board object?